src/System Application/App/Azure File Services API/src/Helper/AFSHttpContentHelper.Codeunit.al

```diff
@@ -0,0 +1,116 @@
+    /// </summary>
+    /// <param name="SourceText">The Text for Request Body.</param>
+    /// <returns>The length of the current stream</returns>
+    [NonDebuggable]
+    procedure GetContentLength(SourceText: Text): Integer
+    var
+        Length: Integer;
+    begin
+        Length := StrLen(SourceText);
+        exit(Length);
+    end;
+
+    [NonDebuggable]
+    procedure AddFilePutContentHeaders(AFSOperationPayload: Codeunit "AFS Operation Payload"; ContentLength: Integer; ContentType: Text; RangeStart: Integer; RangeEnd: Integer)
+    var
+        HttpContent: HttpContent;
+    begin
+        AddFilePutContentHeaders(HttpContent, AFSOperationPayload, ContentLength, ContentType, RangeStart, RangeEnd);
+    end;
+
+    [NonDebuggable]
+    procedure AddFilePutContentHeaders(var HttpContent: HttpContent; AFSOperationPayload: Codeunit "AFS Operation Payload"; var SourceInStream: InStream; RangeStart: Integer; RangeEnd: Integer)
+    var
+        Length: Integer;
+    begin
+        // Do this before calling "GetContentLength", because for some reason the system errors out with "Cannot access a closed Stream."
+        HttpContent.WriteFrom(SourceInStream);
+
+        Length := GetContentLength(SourceInStream);
+
+        AddFilePutContentHeaders(HttpContent, AFSOperationPayload, Length, 'application/octet-stream', RangeStart, RangeEnd);
+    end;
+
+    [NonDebuggable]
+    local procedure AddFilePutContentHeaders(var HttpContent: HttpContent; AFSOperationPayload: Codeunit "AFS Operation Payload"; ContentLength: Integer; ContentType: Text; RangeStart: Integer; RangeEnd: Integer)
+    var
+        Headers: HttpHeaders;
+        FileServiceAPIOperation: Enum "AFS Operation";
+    begin
+        if ContentType = '' then
+            ContentType := 'application/octet-stream';
+
+        HttpContent.GetHeaders(Headers);
+
+        AFSOperationPayload.AddContentHeader('Content-Type', ContentType);
+        if AFSOperationPayload.GetOperation() in [FileServiceAPIOperation::CreateFile] then
+            AFSOperationPayload.AddContentHeader('x-ms-content-length', StrSubstNo(ContentLengthLbl, ContentLength))
+        else
+            AFSOperationPayload.AddContentHeader('Content-Length', StrSubstNo(ContentLengthLbl, ContentLength));
+
+        if AFSOperationPayload.GetOperation() in [FileServiceAPIOperation::PutRange, FileServiceAPIOperation::PutRangefromURL] then begin
+            AFSOperationPayload.AddRequestHeader('x-ms-write', 'update');
+            AFSOperationPayload.AddRequestHeader('x-ms-range', StrSubstNo(RangeLbl, RangeStart, RangeEnd));
+        end;
+    end;
+
+    /// <summary>
+    /// Retrieves the max range avaialble to update with PutRange request.
+    /// </summary>
+    /// <returns>The max range available to update.</returns>
+    [NonDebuggable]
+    procedure GetMaxRange(): Integer
+    begin
+        exit(4 * Power(2, 20));
```

Comment:
What's that number?

Reply:
I guess this should be maximum ChunkSize for File Upload: 4 MiB -> 4194304 Bytes

Reply:
Yes, my intention was for the procedure name to define this value, so this would be max range that can be uploaded at once. I can change it to what pri-kise proposed.

Reply:
A comment describing this "magic" number would be fine :-)

Reply:
But I'll take care of that. I'll push a few updates to this PR in the coming days. Nothing major at all.